src/Microsoft.DotNet.Build.Tasks/VersionTools/MsBuildTraceListener.cs

```diff
@@ -0,0 +1,73 @@
+﻿using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System.Diagnostics;
+using System.Text;
+
+namespace Microsoft.DotNet.Build.Tasks.VersionTools
+{
+    public class MsBuildTraceListener : TraceListener
+    {
+        private TaskLoggingHelper _log;
+        private TraceEventType _eventType;
+        private StringBuilder _partialLine = new StringBuilder();
+
+        public MsBuildTraceListener(TaskLoggingHelper log, TraceEventType eventType)
+        {
+            _log = log;
+            _eventType = eventType;
+
+            Filter = new TraceEventTypeFilter
+            {
+                ShouldTraceType = eventType
+            };
+        }
+
+        public override void Write(string message)
+        {
+            _partialLine.Append(message);
```

Comment:
Is it OK that a dangling "Write" won't get written to the log?  i.e. if Write is the last call, and not WriteLine.